internal/handlers/pg/pgdb/collections.go

```diff
@@ -167,33 +167,25 @@ func CreateCollectionIfNotExists(ctx context.Context, tx pgx.Tx, db, collection
 //
 // TODO Test correctness for concurrent cases https://github.com/FerretDB/FerretDB/issues/1684
 func DropCollection(ctx context.Context, tx pgx.Tx, db, collection string) error {
-	schemaExists, err := schemaExists(ctx, tx, db)
-	if err != nil {
-		return lazyerrors.Error(err)
-	}
-
-	if !schemaExists {
-		return ErrSchemaNotExist
-	}
+	md := newMetadata(tx, db, collection)
+	tableName, err := md.getTableName(ctx)
 
-	table := formatCollectionName(collection)
-	tables, err := tables(ctx, tx, db)
-	if err != nil {
-		return lazyerrors.Error(err)
-	}
-	if !slices.Contains(tables, table) {
+	switch {
+	case err == nil:
+		// do nothing
+	case errors.Is(err, ErrTableNotExist):
 		return ErrTableNotExist
+	default:
+		return lazyerrors.Error(err)
 	}
```

Comment:
can you simplify this? No need to check for `ErrTableNotExist` then return `ErrTableNotExist`
```suggestion
if err != nil{
return err
}
```